sql/catalyst/src/main/java/org/apache/spark/sql/connector/read/PartitionReader.java

```diff
@@ -48,4 +49,12 @@
    * Return the current record. This method should return same value until `next` is called.
    */
   T get();
+
+  /**
+   * Returns an array of custom metrics. By default it returns empty array.
+   */
+  default CustomMetric[] getCustomMetrics() {
```

Comment:
Also, can we split this as a trait, too?

Reply:
I see. I will make it as a trait.